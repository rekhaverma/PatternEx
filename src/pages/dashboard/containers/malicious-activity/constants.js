export const maliciousColumns = [
  {
    'className': 'table__headerColumn',
    'isKey': true,
    'field': 'hash',
    'hidden': true,
    'label': 'Hash',
    'resizable': false,
    'intl': 'malicious.table.hash',
  },
  {
    'className': 'table__headerColumn',
    'isKey': false,
    'field': 'start_time_formatted',
    'hidden': false,
    'label': 'Log Date',
    'resizable': true,
    'width': 100,
    'intl': 'malicious.table.start_time_formatted',
  },
  {
    'className': 'table__headerColumn--entity',
    'isKey': false,
    'field': 'entity_name',
    'hidden': false,
    'label': 'Entity',
    'resizable': true,
    'tooltip': 'investigate',
    'intl': 'malicious.table.entity_name',
  },
  {
    'className': 'table__headerColumn',
    'isKey': false,
    'field': 'score',
    'hidden': false,
    'label': 'Score',
    'resizable': true,
    'width': 150,
    'intl': 'malicious.table.score',
  },
  {
    'className': 'table__headerColumn--highlighted',
    'isKey': false,
    'field': 'threat',
    'hidden': false,
    'label': 'Threat Tactic',
    'resizable': true,
    'width': 220,
    'intl': 'malicious.table.threat',
  },
  {
    'className': 'table__headerColumn--actions',
    'isKey': false,
    'field': 'actions',
    'hidden': false,
    'label': 'Actions',
    'resizable': true,
    'sort': false,
    'width': 250,
    'intl': 'malicious.table.actions',
  },
];
