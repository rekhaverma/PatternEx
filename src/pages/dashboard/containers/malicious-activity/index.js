export { default } from './malicious-activity.container';
