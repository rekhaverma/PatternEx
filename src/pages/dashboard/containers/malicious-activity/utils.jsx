import React from 'react';

import { pipelineToName } from 'lib/decorators';

export const filterByMultipleTags = (data, tags) => {
  if (!tags.length) {
    return data;
  }

  return data.filter((item) => {
    let match = 0;
    tags.forEach((tag) => {
      if (item.entity_name.toLowerCase().includes(tag.toLowerCase())
        || item.threat.toLowerCase().includes(tag.toLowerCase())) {
        match += 1;
      }
    });

    return match === tags.length;
  });
};

export const renderDropdownLabel = (key, pipeline, value) => (
  <span className="dhd-dropdown__option" key={key}>
    <span className="dhd-dropdown__option--label">{pipelineToName(pipeline)}</span>
    <span className="dhd-dropdown__option--value">{value}</span>
  </span>
);
