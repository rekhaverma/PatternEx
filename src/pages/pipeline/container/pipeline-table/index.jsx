export { default } from './pipeline-table.container';
