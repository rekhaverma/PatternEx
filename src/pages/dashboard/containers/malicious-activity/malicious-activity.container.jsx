import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import { connect } from 'react-redux';
import { routerActions } from 'react-router-redux';
import { FormattedMessage } from 'react-intl';

import { enabledPipelines, filterData, filterListData, filtersFromState, isListFiltered } from 'model/selectors';
import { resetFilters, setFilter } from 'model/actions/malicious-activity.actions';
import { tagsI18nMapping } from 'model/classes/heat-map-dropdowns.class';
import { goBackUrl } from 'model/actions/exploded-view.actions';
import { pipelineToName } from 'lib/decorators';

import { Listing } from './components/listing/listing.component';
import HeatMapLegend from '../../components/heatmap-legend';
import { HeatMapDropdown } from './components/heat-map-dropdown/heat-map-dropdown.component';

/**
 * @todo: implement smart table
 * @param props
 * @returns {*}
 * @constructor
 */
export const MaliciousActivity = (props) => {
  const {
    data,
    setLabelForPrediction,
    isOldEVPActive,
    isDataLoaded,
    nextUrl,
    heatMapData,
    filters,
    pipelines,
  } = props;
  const tags = [];
  if (filters.active) {
    Object.keys(filters.active).map((key) => {
      tags.push({
        id: key,
        label: pipelineToName(key),
        removable: false,
      });
      const tacticName = tagsI18nMapping[filters.active[key]];
      if (tacticName && tacticName.name) {
        tags.push({
          id: filters.active[key],
          label: tacticName.name,
          removable: false,
        });
      }
      return key;
    });
  }
  return (
    <section className="dashboard__content +withGradient" style={{ flexWrap: 'wrap' }}>
      <div className="row full-width flex-wrap">
        <HeatMapDropdown
          data={heatMapData}
          filters={filters}
          row={pipelines}
          rowType="pipeline"
          resetFilters={props.resetFilters}
          setFilter={props.setFilter}
        />
        <div className="dashboard__malicious-heatmap">
          <HeatMapLegend />
        </div>
      </div>

      <div className="row full-width">
        <Listing
          searchTags={tags}
          isDataLoaded={isDataLoaded}
          data={data}
          handleExplodedView={props.handleExplodedView}
          deleteLabel={props.deleteLabel}
          setLabelForPrediction={setLabelForPrediction}
          isOldEVPActive={isOldEVPActive}
        />
      </div>
      <div className="row full-width">
        <div className="dashboard__footer">
          <div className="dashboard__toListing">
            <Link to={nextUrl}>
              <span className="icon-chevron-right" />
              <FormattedMessage id="malicious.toMaliciousListing" />
            </Link>
          </div>
        </div>
      </div>
    </section>
  );
};

MaliciousActivity.propTypes = {
  isDataLoaded: PropTypes.bool.isRequired,
  heatMapData: PropTypes.array.isRequired,
  pipelines: PropTypes.array.isRequired,
  data: PropTypes.array.isRequired,
  filters: PropTypes.object.isRequired,
  nextUrl: PropTypes.string.isRequired,
  resetFilters: PropTypes.func.isRequired,
  handleExplodedView: PropTypes.func.isRequired,
  deleteLabel: PropTypes.func.isRequired,
  setLabelForPrediction: PropTypes.func.isRequired,
  setFilter: PropTypes.func.isRequired,
  isOldEVPActive: PropTypes.bool,
};
MaliciousActivity.defaultProps = {
  isOldEVPActive: false,
};

export const mapStateToProps = state => ({
  data: filterListData(state),
  heatMapData: filterData(state),
  isFiltered: isListFiltered(state),
  isDataLoaded: state.raw.getIn(['loadStatus', 'maliciousBehavior']),
  pipelines: enabledPipelines(state),
  filters: filtersFromState(state),
  isOldEVPActive: !state.app.ui.toJS().newEVPVisibility,
});

const mapDispatchToProps = dispatch => ({
  resetFilters: (...args) => dispatch(resetFilters(...args)),
  handleExplodedView: (...args) => dispatch(routerActions.push(...args)),
  setFilter: (...args) => dispatch(setFilter(...args)),
  goBackUrl: (...args) => dispatch(goBackUrl(...args)),
});

export default connect(mapStateToProps, mapDispatchToProps)(MaliciousActivity);
