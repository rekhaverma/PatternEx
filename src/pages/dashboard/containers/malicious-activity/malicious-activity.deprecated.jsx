import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import { routerActions } from 'react-router-redux';
import { connect } from 'react-redux';
import { FormattedMessage } from 'react-intl';

import Loader from 'components/loader/loader-v2.component';
import { resetFilters, setFilter } from 'model/actions/malicious-activity.actions';
import { EvpOpenMethods } from 'model/classes/evp-open-methods';

import {
  filterListData,
  isListFiltered,
  maliciousActivityData,
  filterData,
  filtersFromState,
  enabledPipelines,
} from 'model/selectors';

import AdvancedTable from 'components/advanced-table';
import { DASHBOARD_MALICIOUS_TABLE } from 'config';
import MultiSearch from 'components/multi-search';

import HeatMapContainer from '../heat-map';
import HeatMapLegend from '../../components/heatmap-legend';

import { maliciousColumns } from './constants';

/**
 * @deprecated
 */
class MaliciousActivity extends React.PureComponent {
  constructor(...args) {
    super(...args);

    this.state = {
      tags: [],
    };
    this.handleChange = this.handleChange.bind(this);
  }

  getFilteredData() {
    const { isOldEVPActive, handleExplodedView, data } = this.props;
    let filteredData = data;
    if (this.state.tags.length) {
      filteredData = filteredData.filter((item) => {
        let match = 0;
        this.state.tags.forEach((tag) => {
          if (item.entity_name.toLowerCase().includes(tag.toLowerCase())
            || item.threat.toLowerCase().includes(tag.toLowerCase())) {
            match += 1;
          }
        });

        return match === this.state.tags.length;
      });
    }

    return filteredData.map(el => ({
      ...el,
      'handlers': {
        ...el.handlers,
        'onRowClick': (e, row) => EvpOpenMethods.onRowClickHandler(row, 'malicious', isOldEVPActive, handleExplodedView),
        'onInspect': (e, row) => EvpOpenMethods.onInspectHandler(row, 'malicious', isOldEVPActive),
        'getNewTabUrl': row => EvpOpenMethods.getNewTabUrlHandler(row, 'malicious', isOldEVPActive),
        'onConfirm': (e) => {
          e.stopPropagation();
          const params = {
            ...el,
            start_time: el.end_time,
          };
          this.props.setLabelForPrediction(params, true);
        },
        'onDeny': (e) => {
          e.stopPropagation();
          this.props.setLabelForPrediction(el, false);
        },
        'deleteLabel': (e) => {
          e.stopPropagation();
          if (el.user_tag && el.user_tag_id) { this.props.deleteLabel(el.user_tag.label_id); }
        },
      },
    }));
  }

  handleChange(tags) {
    this.setState({ tags });
  }

  render() {
    const { isDataLoaded, heatMapData, pipelines, nextUrl, filters } = this.props;
    if (!isDataLoaded) {
      return (
        <section className="dashboard__content +withGradient">
          <Loader small />
        </section>
      );
    }

    const filteredData = this.getFilteredData();
    const heatMapContainer = (
      <div className="dashboard__column">
        <span className="dashboard__title">
          <FormattedMessage id="heatmap.title" />
        </span>
        <HeatMapContainer
          data={heatMapData}
          row={pipelines}
          rowType="pipeline"
          setFilter={active => this.props.setFilter('active', active)}
          filters={filters}
        />
        <div className="dashboard__footer">
          <div className="dashboard__toListing">
            <Link to={nextUrl}>
              <span className="icon-chevron-right" />
              <FormattedMessage id="malicious.toMaliciousListing" />
            </Link>
          </div>
          <HeatMapLegend />
        </div>
      </div>
    );

    if (isDataLoaded && filteredData.length === 0) {
      return (
        <section className="dashboard__content +withGradient">
          { heatMapContainer }
          <div className="dashboard__column +noData">
            <h1>There is no data to display</h1>
          </div>
        </section>
      );
    }

    return (
      <section className="dashboard__content +withGradient">
        { heatMapContainer }
        <div className="dashboard__column">
          <AdvancedTable
            classname="table"
            data={filteredData}
            tableConfig={maliciousColumns}
            pageSize="10"
            locationPage={DASHBOARD_MALICIOUS_TABLE}
          >
            <div className="dashboard__topColumn" style={{ 'display': 'flex' }}>
              {
              this.props.isFiltered && (
                <div className="reset" onClick={this.props.resetFilters}>
                  <div className="reset__close">
                    <span className="icon-close" />
                  </div>
                  <span className="reset__filters">Reset Filters</span>
                </div>
              )
            }
              <MultiSearch
                tags={this.state.tags}
                onChange={this.handleChange}
                placeholder="Search for entity or threat tactic"
              />
            </div>
          </AdvancedTable>
        </div>
      </section>
    );
  }
}
MaliciousActivity.propTypes = {
  'heatMapData': PropTypes.array.isRequired,
  'pipelines': PropTypes.array.isRequired,
  'setFilter': PropTypes.func.isRequired,
  'filters': PropTypes.object.isRequired,
  'isFiltered': PropTypes.bool.isRequired,
  'resetFilters': PropTypes.func.isRequired,
  'handleExplodedView': PropTypes.func.isRequired,
  'setLabelForPrediction': PropTypes.func.isRequired,
  'data': PropTypes.array,
  'nextUrl': PropTypes.string,
  'isDataLoaded': PropTypes.bool,
  'isOldEVPActive': PropTypes.bool,
  'deleteLabel': PropTypes.func,
};
MaliciousActivity.defaultProps = {
  'data': [],
  'noData': false,
  'nextUrl': '',
  'isDataLoaded': false,
  'isOldEVPActive': false,
  'deleteLabel': () => {},
};

const mapStateToProps = state => ({
  'data': filterListData(state),
  'heatMapData': filterData(state),
  'noData': maliciousActivityData(state) && maliciousActivityData(state).length === 0,
  'isFiltered': isListFiltered(state),
  'isDataLoaded': state.raw.getIn(['loadStatus', 'maliciousBehavior']),
  'pipelines': enabledPipelines(state),
  'filters': filtersFromState(state),
  'isOldEVPActive': !state.app.ui.toJS().newEVPVisibility,
});

const mapDispatchToProps = dispatch => ({
  'resetFilters': (...args) => dispatch(resetFilters(...args)),
  'handleExplodedView': (...args) => dispatch(routerActions.push(...args)),
  'setFilter': (...args) => dispatch(setFilter(...args)),
});


export default connect(mapStateToProps, mapDispatchToProps)(MaliciousActivity);
