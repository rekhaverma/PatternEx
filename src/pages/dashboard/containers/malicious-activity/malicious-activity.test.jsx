import React from 'react';
import moment from 'moment';
import { MaliciousActivity } from './malicious-activity.container';

/**
 * @status: Complete
 * @sign-off-by: Alex Andries
 */
describe('<MaliciousActivity />', () => {
  it('should match with snapshot', () => {
    const props = {
      isDataLoaded: false,
      data: [],
      heatMapData: [],
      pipelines: [],
      tags: {},
      handleExplodedView: jest.fn(),
      goBackUrl: jest.fn(),
      deleteLabel: jest.fn(),
      setLabelForPrediction: jest.fn(),
      time: {
        startTime: moment.utc('2018-10-11'),
        endTime: moment.utc('2018-10-11'),
        timezone: 'UTC',
      },
      filters: {},
      nextUrl: '',
      resetFilters: jest.fn(),
      setFilter: jest.fn(),
    };
    const wrapper = shallow(<MaliciousActivity {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
