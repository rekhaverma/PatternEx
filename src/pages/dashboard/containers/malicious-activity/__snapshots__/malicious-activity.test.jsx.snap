// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaliciousActivity /> should match with snapshot 1`] = `
<section
  className="dashboard__content +withGradient"
  style={
    Object {
      "flexWrap": "wrap",
    }
  }
>
  <div
    className="row full-width flex-wrap"
  >
    <HeatMapDropdown
      data={Array []}
      filters={Object {}}
      resetFilters={[Function]}
      row={Array []}
      rowType="pipeline"
      setFilter={[Function]}
    />
    <div
      className="dashboard__malicious-heatmap"
    >
      <HeatMapLegend />
    </div>
  </div>
  <div
    className="row full-width"
  >
    <Listing
      data={Array []}
      deleteLabel={[Function]}
      handleExplodedView={[Function]}
      isDataLoaded={false}
      isOldEVPActive={false}
      searchTags={Array []}
      setLabelForPrediction={[Function]}
    />
  </div>
  <div
    className="row full-width"
  >
    <div
      className="dashboard__footer"
    >
      <div
        className="dashboard__toListing"
      >
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to=""
        >
          <span
            className="icon-chevron-right"
          />
          <FormattedMessage
            id="malicious.toMaliciousListing"
            values={Object {}}
          />
        </Link>
      </div>
    </div>
  </div>
</section>
`;
